src/audio/dai.c

```diff
@@ -740,21 +774,23 @@ static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
 		return -EINVAL;
 	}
 
-	if (dd->chan == DMA_CHAN_INVALID)
-		/* get dma channel at first config only */
-		dd->chan = dma_channel_get(dd->dma, channel);
-
-	if (dd->chan < 0) {
-		trace_dai_error_with_ids(dev, "dai_config() error: "
-					 "dma_channel_get() failed");
-		return -EIO;
-	}
+	if (config->type != SOF_DAI_INTEL_HDA) {
+		if (dd->chan == DMA_CHAN_INVALID)
+			/* get dma channel at first config only */
+			dd->chan = dma_channel_get(dd->dma, channel);
 
-	/* set up callback */
-	dma_set_cb(dd->dma, dd->chan, DMA_CB_TYPE_IRQ | DMA_CB_TYPE_COPY,
-		   dai_dma_cb, dev);
+		if (dd->chan < 0) {
+			trace_dai_error_with_ids(dev, "dai_config() error: "
+						 "dma_channel_get() failed");
+			return -EIO;
```

Comment:
Another issue with inconsistent dd->chan value in case of failure.

Reply:
@ranj063 did you address Marcin's comment?

Reply:
@plbossart yes. he has already approved the PR as well.